lib/run/src/Obelisk/Run.hs

```diff
@@ -130,11 +131,11 @@ runWidget
   -> IO ()
 runWidget conf configs frontend validFullEncoder = do
   uri <- either (fail . T.unpack) pure $ getConfigRoute configs
-  let port = fromIntegral $ fromMaybe 80 $ uri ^? uriAuthority . _Right . authPort . _Just
+  let port = fromMaybe (fromIntegral $ fromMaybe 80 $ uri ^? uriAuthority . _Right . authPort . _Just) (_runConfig_portOverride conf)
```

Comment:
Can this be simplified to use like `<|>` or something? Lots of maybe stuff just being thrown around